raiden/ui/web/src/app/components/channel-table/channel-table.component.ts

```diff
@@ -262,12 +252,24 @@ export class ChannelTableComponent implements OnInit {
     }
 
     public handleActivity(eventsParam: EventsParam) {
-        const index = this.watchEvents
-            .indexOf(eventsParam);
-        if (index >= 0 && this.tabIndex - 1 === index) {
-            eventsParam.activity = false;
-        } else {
-            eventsParam.activity = true;
+        const index = this.watchEvents.indexOf(eventsParam);
+        eventsParam.activity = !(index >= 0 && this.tabIndex - 1 === index);
+    }
+
+    private channelIsNotOpen(channel: Channel): boolean {
+        return channel.state !== 'opened';
+    }
+
+    private channelCanTransfer(channel: Channel): boolean {
+        return (channel.state === 'opened' && channel.balance > 0);
+    }
+
+    private updateMenu(channel: Channel, menuItems: MenuItem[]): MenuItem[] {
```

Comment:
Nice approach. May you also, please, remove the initial `disabled` property set in the `menuFor`, and add a call to this method in the end of it, so we can _centralize_ the item availability conditions on this method, instead of having to keep them in sync?